arm-software/linux/docs/GettingStarted.md

```diff
@@ -0,0 +1,322 @@
+#include <stdio.h>
+
+int main()
+{
+  printf("Hello, World!");
+  return 0;
+}
+```
+
+To generate an executable binary, compile your example C program with the
+`armclang` compiler. Specify the input file, `hello.c`, and the binary name
+(using `-o`), `hello`:
+
+```
+$ armclang -o hello hello.c
+```
+
+Run the generated binary `hello`:
+
+```
+$ ./hello
+Hello, World!
+```
+
+#### Example 2: Compile and run example Fortran program
+
+Consider a simple program stored in a `.f90` file, for example: `hello.f90`:
+
+```
+program hello
+  print *, 'hello world'
+end program
+```
+
+To generate an executable binary, compile your example Fortran program with the
+`armflang` compiler. Specify the input file, `hello.f90`, and the binary name
+(using `-o`), `hello`:
+
+```
+$ armflang -o hello hello.f90
+```
+
+Run the generated binary `hello`:
+
+```
+$ ./hello
+ hello world
+```
+
+### Compile and link C/C++ programs
+
+To generate an executable binary, compile your source file (for example,
+`source.c`) with the `armclang` command:
+
+```
+$ armclang source.c
+```
+
+A binary with the filename `a.out` is the output.
+
+Optionally, use the `-o` option to set the binary filename (for example, `binary`):
+
+```
+$ armclang -o binary source.c
+```
+
+You can specify multiple source files on a single line. Each source file is
+compiled individually and then linked into a single executable binary. For
+example, to compile the source files `source1.c` and `source2.c`, use:
+
+```
+$ armclang -o binary source1.c source2.c
+```
+
+To compile each of your source files individually into an object (`.o`) file,
+specify the compile-only option, `-c`, and then pass the resulting object files
+into another invocation of `armclang` to link them into an executable binary:
+
+```
+$ armclang -c source1.c
+
+$ armclang -c source2.c
+
+$ armclang -o binary source1.o source2.o
+```
+
+#### Note
+
+For the C/C++ compiler's command line arguments reference visit this page:
+[Clang command line argument reference](https://releases.llvm.org/20.1.0/tools/clang/docs/ClangCommandLineReference.html).
+
+### Compile and link Fortran programs
+
+To generate an executable binary, compile your source file (for example,
+`source.f90`) with the `armflang` command:
+
+```
+$ armflang source.f90
+```
+
+A binary with the filename `a.out` is the output.
+
+You can specify multiple source files on a single line. Each source file is
+compiled individually and then linked into a single executable binary. For
+example, to compile the source files `source1.f90` and `source2.f90`, use:
+
+```
+$ armflang -o binary source1.f90 source2.f90
+```
+
+To compile each of your source files individually into an object (`.o`) file,
+specify the compile-only option, `-c`, and then pass the resulting object files
+into another invocation of `armflang` to link them into an executable binary:
+
+```
+$ armflang -c source1.f90
+
+$ armflang -c source2.f90
+
+$ armflang -o binary source1.o source2.o
+```
+
```

Comment:
Maybe it would be useful to mention that if you have several object files from multiple languages, it is *very* strongly encouraged to link using armflang and not armclang.

Reply:
Added, @tblah could you look at it?